File: emulinker/src/main/java/org/emulinker/util/HexConverter.java
package org.emulinker.util;

import static java.nio.charset.StandardCharsets.UTF_8;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;

public class HexConverter {
  public static void main(String args[]) throws Exception {
    BufferedOutputStream os = new BufferedOutputStream(new FileOutputStream(args[1]));
    BufferedReader is = Files.newBufferedReader(Paths.get(args[0]), UTF_8);
    String line = null;
    while ((line = is.readLine()) != null) {
      byte[] bytes = EmuUtil.hexToByteArray(line);
      os.write(bytes);
    }

    is.close();
    os.close();
  }
}

File: emulinker/src/main/java/org/emulinker/util/HexConverter.kt
package org.emulinker.util

import java.io.BufferedOutputStream
import java.io.FileOutputStream
import java.lang.Exception
import java.nio.charset.StandardCharsets
import java.nio.file.Files
import java.nio.file.Paths
import kotlin.Throws
import kotlin.jvm.JvmStatic

object HexConverter {
  @Throws(Exception::class)
  @JvmStatic
  fun main(args: Array<String>) {
    val os = BufferedOutputStream(FileOutputStream(args[1]))
    val `is` = Files.newBufferedReader(Paths.get(args[0]), StandardCharsets.UTF_8)
    var line: String? = null
    `is`.forEachLine {
      val bytes = EmuUtil.hexToByteArray(it)
      os.write(bytes)
    }
    `is`.close()
    os.close()
  }
}
